include/begin_code.h

```diff
@@ -185,3 +185,33 @@
 #undef _HAS_FALLTHROUGH
 #endif /* C++17 or C2x */
 #endif /* SDL_FALLTHROUGH not defined */
+
+#ifndef SDL_MALLOC
+#if defined(__GNUC__)
+#define SDL_MALLOC __attribute__((malloc))
+#elif defined(_MSC_VER)
+#define SDL_MALLOC__desclspec(restrict)
```

Comment:
Additionally, MSVC has an [allocator declaration specifier](https://docs.microsoft.com/nl-be/cpp/cpp/allocator?view=msvc-160), which is used for profiling and memory tracking.
```suggestion
#define SDL_MALLOC__desclspec(restrict) __declspec(allocator)
```
I think (!) both attributes can also be added to all other `SDL_ALLOC_SIZE`'s.

Reply:
I know but I was wary about VS versions supported.

Reply:
> I think (!) both attributes can also be added to all other `SDL_ALLOC_SIZE`'s.

not realloc